graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -709,6 +710,40 @@ def _preprocess_graphql_string(graphql_string):
     return graphql_string + '\n'
 
 
+def _validate_schema_and_ast(schema, ast):
+    """Validate the supplied graphql schema and ast.
+
+    This method wraps around graphql-core's validation to enforce a stricter requirement of the
+    schema-- all directives supported by the compiler must be declared by the schema, regardless of
+    whether each directive is used in the query or not.
+
+    Args:
+        schema: GraphQL schema object, created using the GraphQL library
+        ast: abstract syntax tree representation of a graphql query
+
+    Returns:
+        list containing schema and/or query validation errors
+    """
+    core_graphql_errors = validate(schema, ast)
+
+    # Extract name, locations, and args keys in order to compare schema directives and directives
+    # which are supported by the graphql compiler.
+    graphql_compiler_directives = [(s.name, s.locations, s.args.keys()) for s in DIRECTIVES]
+
+    # Directives provided in the schema.
+    # pylint: disable=protected-access
+    schema_directives = [
+        (directive.name, directive.locations, directive.args.keys())
+        for directive in schema._directives
+    ]
+    # pylint: enable=protected-access
+    for directive in graphql_compiler_directives:
```

Comment:
Seems like we're trying to compare two sets of declared directives for equality. Thoughts on doing that explicitly?

Specifically, I'm thinking we could make the two lists of directives (expected vs actual) into sets. Then, we can compute:
```
missing_directives = expected_directives - actual_directives
unexpected_directives = actual_directives - expected_directives
```
If either of these two sets is non-empty, we can complain with an appropriate error message and say specifically which directives are the issue.